algorithms/kmp_search.cr

```diff
@@ -0,0 +1,36 @@
+module Algorithms::Search
+  def self.kmp_table(patt : String)
+    tbl = Array.new(patt.size + 1, -1)
+    k = -1
+    i = 1
+    while i <= patt.size
```

Comment:
Could we use a range here? (Something like `(1..patt.size).each { |i| ... }`)

Reply:
Absolutely, thanks for the suggestion!